wrangler-proto/src/main/java/io/cdap/wrangler/proto/workspace/v2/Workspace.java

```diff
@@ -79,6 +84,15 @@ public JsonObject getInsights() {
     return insights;
   }
 
+  public HashMap<String, UserDefinedAction> getNullabilityMap() {
+    return nullabilityMap;
+  }
+
+  public void setNullabilityMap(
```

Comment:
should usually not have any setter methods. Making the state mutable makes it very difficult to reason about correctness. If it needs to be changed, a new instance of the Workspace should be created using the Builder.

Reply:
I see that all the other fields have setters and the Builder itself uses setter method, so even if I am creating a new instance using Builder I'll still need the setter method. How will changing the entire workspace using the Builder that uses the setter method be different from using the setter method directly?